docs/source/opkg-keyrings/opkg-keyrings_index.rst

```diff
@@ -0,0 +1,157 @@
+========================================================
+NI Linux Real-Time and opkg-keyrings: Upgrading keyrings
+========================================================
+
+.. contents:: Table of Contents
+   :depth: 2
+   :local: 
+
+Introduction
+============
+
+The opkg-keyrings package is a critical component in systems using opkg for
+package management. It contains a collection of public signing keys used to
+verify the authenticity and integrity of packages downloaded from a repository or feed.
+
+The opkg-keyrings package is updated periodically to add new keys or remove
+expired keys. It is important to keep the keyrings up-to-date to ensure that
+packages can be verified and installed correctly.
+
+This document provides instructions on how to upgrade the opkg-keyrings
+package on NI Linux Real-Time systems.
+
+Purpose of the opkg-keyrings package
+-------------------------------------
+#. Store Trusted Signing Keys:
+
+    * The package includes the public keys required to verify .ipk package signatures.
+    * These keys are placed in specific directories, such as /usr/share/opkg/keyrings/.
+
+#. Ensure Package Integrity:
+
+    * During installation, opkg checks package signatures using these keys.
+    * This prevents unauthorized or tampered packages from being installed.
+
+#. Key Management:
+
+    * The opkg-keyrings package is updated when signing keys are rotated (e.g., when
+        old keys are retired, and new ones are issued).
+    * Upgrading this package ensures the keyring remains current and secure.
+
+How opkg signing works and why we do it
+=======================================
+opkg signing is a mechanism to ensure the authenticity and integrity of .ipk
+packages being installed on a system. It involves cryptographic signatures that
+verify packages come from a trusted source and have not been tampered with. 
+
+How does it work?
+-----------------
+When a package is built, it is signed with a private key. The corresponding
+public key is included in the opkg-keyrings package. When the package is
+downloaded, opkg checks the package signature against the public key to
+verify the package's authenticity and integrity.
+
+Steps involved:
+----------------
+
+#. Generate a Signing Key:
+
+    * A pair of private and public keys is created using a tool like gpg.
+    * The private key is used to sign the package index or individual packages
+        (package index, the *Packages* and *Packages.gz* files in the case of NILRT).
+    * The public key is distributed to the target systems.
+
+#. Sign the Package Index:
+
+    * The package index (*Packages* file) lists all available packages and their
```

Comment:
This wording makes it sound like the Packages file itself contains the signatures.. better to clarify.

Reply:
I have rephrased this.